shopinvader_sale_coupon/services/abstract_sale.py

```diff
@@ -15,21 +15,15 @@ def _convert_one_sale(self, sale):
         res = super()._convert_one_sale(sale)
         res.update(
             {
-                "promo_code": sale.promo_code,
+                "promo_codes": sale.promo_codes,
                 "reward_amount": sale.reward_amount,
                 "reward_amount_tax_incl": sale.reward_amount_tax_incl,
                 "applied_coupon_ids": self._convert_coupon(sale.applied_coupon_ids),
                 "generated_coupon_ids": self._convert_coupon(sale.generated_coupon_ids),
-                "no_code_promo_program_ids": self._convert_coupon_programs(
-                    sale.no_code_promo_program_ids
-                ),
-                "code_promo_program_id": (
-                    sale.code_promo_program_id.jsonify(
-                        self._parser_coupon_program(), one=True
-                    )
-                    if sale.code_promo_program_id
-                    else False
+                "no_code_promo_program_ids": self._convert_program(
+                    sale.no_code_program_ids
                 ),
+                "code_promo_program_ids": self._convert_program(sale.code_program_ids),
```

Comment:
In v16 several programs with code can be applied on the same SO so the schema structure had to be changed.